src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceComponent.scala

```diff
@@ -202,29 +206,69 @@ trait WorkspaceComponent {
         val flatRealms = allRealms.flatten.toSet
         DBIO.sequence(flatRealms.toSeq.map { realm =>
-          rawlsGroupQuery.loadGroupIfMember(realm, user) flatMap {
+          val groupRef = RawlsGroupRef(RawlsGroupName(realm))
+          rawlsGroupQuery.loadGroupIfMember(groupRef, user) flatMap {
             case None => DBIO.successful(None)
-            case Some(_) => DBIO.successful(Some(realm))
+            case Some(_) => DBIO.successful(Some(groupRef))
           }
         })
       }
     }
 
-    private def workspaceAttributes(workspaceId: UUID) = for {
-      workspaceAttrRec <- workspaceAttributeQuery if workspaceAttrRec.workspaceId === workspaceId
+    def listAccessGroupMemberEmails(workspaceIds: Seq[UUID], accessLevel: WorkspaceAccessLevel): ReadAction[Map[UUID, Seq[String]]] = {
+      val accessQuery = workspaceAccessQuery.filter(access => access.workspaceId.inSetBind(workspaceIds) && access.isRealmAcl === false && access.accessLevel === accessLevel.toString)
+
+      val subGroupEmailQuery = accessQuery join rawlsGroupQuery.subGroupEmailsQuery on (_.groupName === _._1) map (r => (r._1.workspaceId, r._2._2))
+      val userEmailQuery = accessQuery join rawlsGroupQuery.userEmailsQuery on (_.groupName === _._1) map (r => (r._1.workspaceId, r._2._2))
+
+      (subGroupEmailQuery union userEmailQuery).result.map { _.groupBy(_._1).mapValues(_.map(_._2)) }
+    }
+
+    def listSubmissionSummaryStats(workspaceIds: Seq[UUID]): ReadAction[Map[UUID, WorkspaceSubmissionStats]] = {
+      val workflowDatesQuery = (for {
+        submissions <- submissionQuery if submissions.workspaceId.inSetBind(workspaceIds)
+        workflows <- workflowQuery if submissions.id === workflows.submissionId
+      } yield (submissions.workspaceId, workflows.status, workflows.statusLastChangedDate)).
+        groupBy(r => (r._1, r._2)).
+        map { case ((wsId, wfStatus), records) => (wsId, wfStatus, records.map(_._3).max) }
+
+      val runningSubmissionsQuery = (for {
+        submissions <- submissionQuery if submissions.workspaceId.inSetBind(workspaceIds) && submissions.status === SubmissionStatuses.Submitted.toString
+      } yield submissions).groupBy(_.workspaceId).map { case (wfId, submissions) => (wfId, submissions.length)}
+
+      for {
+        workflowDates <- workflowDatesQuery.result
+        runningSubmissions <- runningSubmissionsQuery.result
+      } yield {
+        val mappedWorkflowDates = workflowDates.groupBy(_._1).mapValues(_.groupBy(_._2).mapValues(_.head._3))
+        val mappedSubmissions = runningSubmissions.groupBy(_._1).mapValues(_.head._2)
+
+        workspaceIds.map { wsId =>
+          val (lastFailedDate, lastSuccessDate) = mappedWorkflowDates.get(wsId) match {
+            case None => (None, None)
+            case Some(datesByStatus) => (datesByStatus.getOrElse(WorkflowStatuses.Failed.toString, None), datesByStatus.getOrElse(WorkflowStatuses.Succeeded.toString, None))
```

Comment:
not quite, the value in the map is itself an option so just doing get results in Option[Option[...]]